rewrite-maven/src/main/java/org/openrewrite/maven/RemoveUnneededDependencyOverrides.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.tree.DependencyManagementDependency;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.maven.tree.Pom;
+import org.openrewrite.xml.RemoveContentVisitor;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Collection;
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveUnneededDependencyOverrides extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Remove dependency overrides in descendant poms";
+    }
+
+    @Override
+    public String getDescription() {
+        return "For child poms that contain the same dependency version as an ancestor," +
+                " remove the explicit version in the child.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new FindManagedDependencyVersionVisitor();
+    }
+
+    private static class FindManagedDependencyVersionVisitor extends MavenVisitor {
+        @Override
+        public Maven visitMaven(Maven maven, ExecutionContext ctx) {
+            Collection<Pom.Dependency> dependencies = maven.getModel().getDependencies();
+            for (Pom.Dependency dependency : dependencies) {
+                String managedVersion = findManagedVersion(maven.getModel(), dependency);
```

Comment:
I believe I'm going to need to add scope checking here too -- will add another couple tests for that.

Reply:
Fixed in newest revision.